# How common are low-complexity domains in different types of organisms?

Proteins are sequences of building blocks called "amino acids". Protein sequences are represented as strings of letters, where each letter represents a different amino acid. There are 20 different types of amino acids. Since there are 20<sup>N</sup> possible combinations of amino acids for a sequence of length *N*, this can lead to very complex and a lot of combinations! Despite this possibility, some protein parts are very simple sequences, meaning they are comprised mostly of only one or two amino acids. For comparison, a complex sequence might look something like "ADGMNTRQSTNGPAHIVMNL", whereas a simple sequence might look something like "NNQNNQYQNQNYYQNQQQNNQNN". In the first case, there is a complex mixture of amino acids, whereas in the second case the entire sequence consists of only three amino acids (N, Q, and Y).

These simple sequences, or LCDs, are biologically interesting because they often have unusual chemical properties and behaviors in living organisms. However, no one had yet discovered how common each type of LCD is and the functions of each type of LCD in different organisms on a large scale. In this study, I analyzed the frequency and functions of LCDs in all proteins from >21,000 organisms. The figure below highlights just one of the techniques that I developed to quantify how common each type of LCD is in an individual organism compared to a "null" expectation based on how common each amino acid is in that organism.


## Brief Summary
- **Panel A:** Certain types of LCDs (bright colors) are common in the malaria proteome. This heatmap shows the natural log of the odds ratio for each type of LCD compared to a randomized version of the Malaria proteome. The "Primary Amino Acid" (x-axis) essentially represents the most prevalent amino acid in the LCD, whereas the "Secondary Amino Acid" (y-axis). Together, these two amino acids comprise the "LCD class".
- **Panel B:** Indication of statistical significance for LCD enrichment according to each LCD class. Red squares indicate that the observed LCD enrichment is statistically significant.
- **Panel C:** Same as Panel A, but for the human proteome. Notice that different classes of LCDs are enriched in the human proteome compared to the malaria proteome.
- **Panel D:** Same as Panel B, but for the human proteome. Notice that more classes and different classes of LCDs are statistically significant compared to the malaria proteome.
- **Key discoveries:** LCD enrichment differs between organisms and depends on LCD class. Statistically significant LCD enrichment can occur even when the amino acids are extremely common in the proteome (e.g., N, D, and E LCDs in the malaria proteome).
- **Python code for analyses and data visualizations for all Panels are included in this repository.** Note that the "compare_Observed_vs_Scrambled_Frequencies.py" script creates a file that is required to run the "plot_IndividualOrganism_lnORs_and_Pvals.py" script, so they must be run sequentially.

![LCD figure](https://github.com/seancascarina/One_Figure_Summaries/blob/main/2024_PLOS_Comput_Biol/LCD_Frequencies_Malaria_and_Humans.png)

## Additional Detail
**Panels A and C** show the degree of LCD enrichment in malaria and humans for each LCD class. These heatmaps were created by performing separate searches for each LCD class in the given organism (either malaria or humans) using a custom search algorithm that I built called "LCD-Composer". For each LCD class, the number of proteins containing that type of LCD was counted. Then, a randomized version of the proteome was created, where the order of amino acids was completely shuffled. This accounts for differences in whole-proteome amino acid frequencies: just by chance, more common amino acids are expected to result in more instances of LCDs containing that amino acid. LCD searches were repeated on these randomized versions of the proteomes, and LCD frequencies were counted again. LCD enrichment was then calculated as the natural log of the odds ratio (lnOR) of the number of proteins containing an LCD from the original proteome compared to the number of proteins containing an LCD from the randomized proteome. These lnOR values are depicted in the heatmaps.

**Panels B and D** show label-encoded statistical significance indicators showing whether the LCD enrichment for each LCD class is statistically significant or not. *P*-values were calculated using Fisher's exact test, and a multiple test correction (Sidak) was performed to control the Type I error rate. These *p*-values were then compared to the pre-defined *p*<0.05 threshold to generate the label encoding: *p*-values less than 0.05 were assigned to the red class, whereas *p*-values greater than 0.05 were assigned to the black class. The grey squares indicate LCD classes that were completely absent from the original proteome and the corresponding randomized proteome, which would make the statistical comparisons uninformative.